pkg/descheduler/descheduler.go

```diff
@@ -80,6 +80,13 @@ func Run(ctx context.Context, rs *options.DeschedulerServer) error {
 
 type strategyFunction func(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor, getPodsAssignedToNode podutil.GetPodsAssignedToNodeFunc)
 
+type Strategy interface {
```

Comment:
This actually align with the effort of moving to plugins. During the "descheduling framework" we would first initialize and validate all strategies.

Reply:
i can close this pr and hold off till a proposal is defined if you'd like.

Reply:
I think let's keep it open for now so it's easier to find. Thanks for doing this @jklaw90, I think this has some big picture potential to improve the descheduler

I'd like to start gathering some feedback on what a descheduling framework would look like, so I'll open an issue to get some discussion going on that and include this as an example.